src/mem3/src/mem3.erl

```diff
@@ -198,6 +199,20 @@ shard_suffix(DbName0) when is_binary(DbName0) ->
 shard_suffix(Db) ->
     shard_suffix(couch_db:name(Db)).
 
+shard_creation_time(DbName0) ->
+    Shard = hd(shards(DbName0)),
+    case Shard#shard.name of
+        <<"shards/", _:8/binary, "-", _:8/binary, "/", DbName/binary>> ->
+            case filename:extension(binary_to_list(DbName)) of
```

Comment:
We might be able to avoid binary_to_list then back to binary:

```
1> DotExt = filename:extension(<<"a/b/c.ext">>).
<<".ext">>
2> <<".", JustExt/binary>> = DotExt.
<<".ext">>
3> JustExt.
<<"ext">>
4>
```

Tested on Erlang 20